### PR TITLE
Improve code quality

### PR DESCRIPTION
**Pull request by yixiann**
Main changes were to add javadocs to classes and methods.  

**Some markdown files did not have a line break before a list**
image

**Some classes had overly long methods / Constructors:**
image

ItemCard constructor
EditCommand constructor
JsonAdaptedItem toModelType()

SLAP was applied to break them down into simple components. 

**Empty author tag was added to:**
UserPrefs
Version
SampleDataUtil
ItemsToSort

**Some public fields were converted to private fields:**
ItemBuilder

**Added assertions to Item getters"

### CODE REVIEW

File: src/main/java/seedu/foodrem/model/item/ItemDate.java [hunk 1]
```diff
@@ -60,8 +69,8 @@ public boolean isAfterDate(ItemDate itemDate) {
     @Override
     public boolean equals(Object other) {
         return other == this
-                && other instanceof ItemDate
-                && date.equals(((ItemDate) other).date);
+                || (other instanceof ItemDate
```
Comment (RichDom2185) [inline src/main/java/seedu/foodrem/model/item/ItemDate.java:72]: Thanks for spotting this bug!

File: src/main/java/seedu/foodrem/model/item/itemvalidators/ItemPriceValidator.java [hunk 2]
```diff
@@ -26,7 +26,10 @@ public static Void validate(String itemPriceString) {
                 "The item price should not be negative.");
     }
 
-    static Void validateNumericString(String numericString, int maxDecimalPlace, int maximum,
+    /**
+     * Validates a given numeric String. This is to be used during construction.
+     */
+    public static Void validateNumericString(String numericString, int maxDecimalPlace, int maximum,
```
Comment (RichDom2185) [inline src/main/java/seedu/foodrem/model/item/itemvalidators/ItemPriceValidator.java:32]: Why was this changed to public?

**Review by RichDom2185 — APPROVED** [review]
LGTM, just a few questions, thanks! Also, what's the empty `@author` for?

File: src/main/java/seedu/foodrem/model/item/itemvalidators/ItemPriceValidator.java (hunk 2 above)
Comment (yixiann) [reply src/main/java/seedu/foodrem/model/item/itemvalidators/ItemPriceValidator.java]: Sorry, I was a mistake

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!
